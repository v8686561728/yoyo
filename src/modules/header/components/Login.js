import React, { useState } from "react";
// import history from '../../common/components/history';
import { login } from "../state/actions/index";
import { GoogleLogin } from "react-google-login";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import CssBaseline from "@material-ui/core/CssBaseline";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { google } from "../../../config/constants";
import { connect } from "react-redux";

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main
  },
  form: {
    // width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  submit: {
    margin: theme.spacing(3, 0, 2)
  },
  mainContainer: {
    width: 300
  }
}));

export function Login({ login }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const classes = useStyles();

  const responseGoogle = response => {
    login(response);
  };
  const loginSubmit = e => {
    e.preventDefault();
    let data = { email: email, password: password };
    login(data);
  };
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justify="center"
      style={{ minHeight: "100vh" }}
    >
      <Grid item xs={10}>
        <div component="main" className={classes.mainContainer}>
          <CssBaseline />
          <div className={classes.paper}>
            <Avatar className={classes.avatar}>
              <LockOutlinedIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
              Sign in
            </Typography>
            <form className={classes.form} noValidate>
              <TextField
                variant="outlined"
                margin="normal"
               
                fullWidth
                id="email"
                label="Email Address"
                name="email"
                autoComplete="email"
                autoFocus
                required="true"
                onChange={e => {
                  setEmail(e.target.value);
                }}
                
              />
              <TextField
                variant="outlined"
                margin="normal"
            
                fullWidth
                required="true"
                name="password"
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                onChange={e => {
                  setPassword(e.target.value);
                }}
              />
              <FormControlLabel
                control={<Checkbox value="remember" color="primary" />}
                label="Remember me"
              />
              <Button
                id="xyz"
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
                onClick={e => loginSubmit(e)}
              >
                Sign In
              </Button>
              <GoogleLogin
                className={classes.submit}
                clientId={google.loginClientId}
                buttonText="Login with Google"
                onSuccess={responseGoogle}
                onFailure={responseGoogle}
                cookiePolicy={"single_host_origin"}
              />

              <Grid container>
                <Grid item xs>
                  <Link href="#" variant="body2">
                    Forgot password?
                  </Link>
                </Grid>
                <Grid item>
                  <Link href="#" variant="body2">
                    {"Don't have an account? Sign Up"}
                  </Link>
                </Grid>
              </Grid>
            </form>
          </div>
          <Box mt={8}></Box>
        </div>
      </Grid>
    </Grid>
  );
}

export default connect(null, { login })(Login);
