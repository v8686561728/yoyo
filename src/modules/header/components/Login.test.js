import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Login } from "./Login";

let props = {
  //   e: {
  //     preventDefault: jest.fn()
  //   }
  login: jest.fn()
};
it("should render Login", () => {
  let wrapper = shallow(<Login {...props} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should trigger onSubmit on click ", () => {
  let wrapper = shallow(<Login {...props} />);
  wrapper.find("#xyz").simulate("click", { preventDefault: () => {} });
});
