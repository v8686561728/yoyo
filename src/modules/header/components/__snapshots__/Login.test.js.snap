// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  spacing={0}
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <div
      className="makeStyles-mainContainer-5"
      component="main"
    >
      <CssBaseline />
      <div
        className="makeStyles-paper-1"
      >
        <WithStyles(ForwardRef(Avatar))
          className="makeStyles-avatar-2"
        >
          <pure(LockOutlinedIcon) />
        </WithStyles(ForwardRef(Avatar))>
        <WithStyles(ForwardRef(Typography))
          component="h1"
          variant="h5"
        >
          Sign in
        </WithStyles(ForwardRef(Typography))>
        <form
          className="makeStyles-form-3"
          noValidate={true}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="email"
            autoFocus={true}
            fullWidth={true}
            id="email"
            label="Email Address"
            margin="normal"
            name="email"
            onChange={[Function]}
            required={true}
            variant="outlined"
          />
          <WithStyles(ForwardRef(TextField))
            autoComplete="current-password"
            fullWidth={true}
            id="password"
            label="Password"
            margin="normal"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            variant="outlined"
          />
          <WithStyles(ForwardRef(FormControlLabel))
            control={
              <ForwardRef(WithStyles)
                color="primary"
                value="remember"
              />
            }
            label="Remember me"
          />
          <WithStyles(ForwardRef(Button))
            className="makeStyles-submit-4"
            color="primary"
            fullWidth={true}
            id="xyz"
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Sign In
          </WithStyles(ForwardRef(Button))>
          <u
            accessType="online"
            buttonText="Login with Google"
            className="makeStyles-submit-4"
            clientId="170084332667-mfcodrlla6eq1mkqu68kf4289sl11pe6.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            icon={true}
            isSignedIn={false}
            jsSrc="https://apis.google.com/js/api.js"
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            scope="profile email"
            tag="button"
            theme="light"
            type="button"
            uxMode="popup"
          />
          <WithStyles(ForwardRef(Grid))
            container={true}
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
              xs={true}
            >
              <WithStyles(ForwardRef(Link))
                href="#"
                variant="body2"
              >
                Forgot password?
              </WithStyles(ForwardRef(Link))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Link))
                href="#"
                variant="body2"
              >
                Don't have an account? Sign Up
              </WithStyles(ForwardRef(Link))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </form>
      </div>
      <Styled(MuiBox)
        mt={8}
      />
    </div>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
